x/authority/keeper/msg_server_update_chain_info.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	cosmoserror "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/zetacore/x/authority/types"
+)
+
+// UpdateChainInfo updates the chain inffo structure that adds new static chain info or overwrite existing chain info
+// on the hard-coded chain info
+func (k msgServer) UpdateChainInfo(
+	goCtx context.Context,
+	msg *types.MsgUpdateChainInfo,
+) (*types.MsgUpdateChainInfoResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// This message is only allowed to be called by group admin
+	// Group admin because this functionality would rarely be called
+	// and overwriting false chain info can have undesired effects
```

Comment:
NIT: imo , this is a bit extra info, we don't add reasoning for the other admin messages on zetacore. 
It would be good to collect this information in our notion doc for policies though.

Reply:
I would leave it for now